data/scenarios/Tutorials/loops.yaml

```diff
@@ -0,0 +1,77 @@
+version: 1
+name: Loops
+description: |
+  Learn how to implement loops.
+objectives:
+  - goal:
+      - |
+        The previous tutorial taught you how to use conditional
+        expressions; this one will guide you through using loops.
+      - |
+        To the east of your base lies a path of `rock`{=entity}s.  The
+        path extends in a straight line for an unknown distance, and at
+        the end of the path lies a lump of `gold`{=entity}.  You must
+        program a robot to go and fetch it!
+      - |
+        Some kind of `while`{=snippet} loop would obviously come in
+        handy here. However, the Swarm language does not have loops,
+        only recursion, which is enabled by the `strange loop`{=entity}
+        device.  You can use recursion to implement your own control
+        structures.  For example, try filling in the `...`{=snippet}
+        below to implement your own while loop function:
+      - |
+        `def while : Cmd Bool -> Cmd Unit -> Cmd Unit = \cond. \body. ... end`{=snippet}
```

Comment:
I did not realize there wasn't a recursion tutorial. 😲 

Could you maybe add a small recursion example before the `while`? Maybe I am making it too easy, but I would add:

````markdown
For example a program that repeats forever:
```
def forever : Cmd Unit -> Cmd Unit = \body.
  body; forever body
end
```
Now try filling in the `...`{=snippet} below to implement your own while loop function
to lead you along the path of rocks.
````